org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ScalaCompileUnits.scala

```diff
@@ -0,0 +1,89 @@
+package org.scalaide.core.internal.project
+
+import scala.tools.nsc.Settings
+
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.core.runtime.SubMonitor
+import org.scalaide.core.IScalaProject
+import org.scalaide.core.internal.builder.EclipseBuildManager
+import org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager
+import org.scalaide.core.internal.project.scopes.BuildMacrosUnit
+import org.scalaide.core.internal.project.scopes.BuildMainUnit
+import org.scalaide.core.internal.project.scopes.BuildScopeUnit
+import org.scalaide.core.internal.project.scopes.BuildTestsUnit
+import org.scalaide.core.internal.project.scopes.CompileMacrosScope
+import org.scalaide.core.internal.project.scopes.CompileMainScope
+import org.scalaide.core.internal.project.scopes.CompileTestsScope
+
+trait CompileScope {
+  def name: String
+}
+
+trait DependenciesTrackable {
+  def canTrackDependencies: Boolean
+}
+
+trait BuildManagerFactory extends DependenciesTrackable {
+  def buildManager(owningProject: IScalaProject, managerSettings: Settings): EclipseBuildManager
+}
+
+trait EclipseSbtBuildManagerFactory extends BuildManagerFactory {
+  def buildManager(owningProject: IScalaProject, managerSettings: Settings): EclipseBuildManager =
+    new EclipseSbtBuildManager(owningProject, managerSettings)
+  def canTrackDependencies = true
+}
+
+trait BuildScopeUnitsManager extends DependenciesTrackable {
```

Comment:
Could this extends the BuildManager? The interface looks quite similar, with a couple of additional methods.